nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java

```diff
@@ -0,0 +1,451 @@
+            if (colIdx != -1) {
+                ColumnSchema colSchema = schema.getColumnByIndex(colIdx);
+                Type colType = colSchema.getType();
+
+                if (record.getValue(colName) == null) {
+                    if (schema.getColumnByIndex(colIdx).isKey()) {
+                        throw new IllegalArgumentException(String.format("Can't set primary key column %s to null ", colName));
+                    } else if(!schema.getColumnByIndex(colIdx).isNullable()) {
+                        throw new IllegalArgumentException(String.format("Can't set primary key column %s to null ", colName));
+                    }
+
+                    if (!ignoreNull) {
+                        row.setNull(colName);
+                        continue;
+                    }
+                } else {
+                    switch (colType.getDataType(colSchema.getTypeAttributes())) {
+                        case BOOL:
+                            row.addBoolean(colIdx, record.getAsBoolean(colName));
+                            break;
+                        case FLOAT:
+                            row.addFloat(colIdx, record.getAsFloat(colName));
+                            break;
+                        case DOUBLE:
+                            row.addDouble(colIdx, record.getAsDouble(colName));
+                            break;
+                        case BINARY:
+                            row.addBinary(colIdx, record.getAsString(colName).getBytes());
+                            break;
+                        case INT8:
+                            row.addByte(colIdx, record.getAsInt(colName).byteValue());
+                            break;
+                        case INT16:
+                            row.addShort(colIdx, record.getAsInt(colName).shortValue());
+                            break;
+                        case INT32:
+                            row.addInt(colIdx, record.getAsInt(colName));
+                            break;
+                        case INT64:
+                        case UNIXTIME_MICROS:
+                            row.addLong(colIdx, record.getAsLong(colName));
+                            break;
+                        case STRING:
+                            row.addString(colIdx, record.getAsString(colName));
+                            break;
+                        case DECIMAL32:
+                        case DECIMAL64:
+                        case DECIMAL128:
+                            row.addDecimal(colIdx, new BigDecimal(record.getAsString(colName)));
+                            break;
+                        default:
+                            throw new IllegalStateException(String.format("unknown column type %s", colType));
+                    }
+                }
+            }
+        }
+    }
+
+    private void addPredicate(AsyncKuduScanner.AsyncKuduScannerBuilder scannerBuilder, KuduTable kuduTable, String column, String value, String comparisonOp) {
```

Comment:
Kudu 1.10 has a generic addPredicate that can take an object. That might simplify things here.